sirepo/job_supervisor.py

```diff
@@ -327,14 +321,24 @@ def _ops_to_cancel():
                 # Do not cancel sim frames. Allow them to come back for a cancelled run
                 if not (self.db.isParallel and o.opName == job.OP_ANALYSIS)
             ]
-            if op and op in self.ops:
+            if op and op in self.ops and op not in o:
                 o.append(op)
             return o
 
         r = PKDict(state=job.CANCELED)
         if (
-            not self._req_is_valid(req)
-            or self.db.status not in _RUNNING_PENDING
+# TODO(e-carlin): the 'and not op' check is not quite right. The problem is if
+# we are calling api_runCancel from run_timeout and the op that timed out was
+# an analysis op that is not normally checked for request validity. Ex
+# api_downloadDataFile does not check for a valid request because a data file
+# request is not valid (it has not computeJobHash for example). But we still
+# need to cancel downloadDataFile ops. We know we are called from run_timeout
+# if op is not None so that is how we are getting around the problem. This
+# could result in odd UX. Ex a running simulation being cancelled due to a
+# downloadDataFile request timeout in another browser window (only the
+# computJids must match between the two requests).
```

Comment:
Getting complicated. Let's talk about this in person. I want to understand completely.